### DISCUSSION TITLE
problema com o componete http via wifi

### DISCUSSION THREAD

**Discussion by 30103010** [Flutterando/forum#137] [Perguntas e Respostas]
Ola pessoal , venho pedir ajuda a voces porque nao sei mais o que fazer com esse problema.

eu tenho uma solicitacao sendo feita para uma api com o  (http: ^0.13.5)

`var response = await http
          .get(
        Uri.parse(link),
        headers: headers,
   )`
o link é um link http:\\ e nao https:\\

a conexao fica travada, pensando pensando a vida toda, sem retornar nada , nenhum erro. 
tentei fazer a solicitacao pelo navegador do celular e é feita normalmente.

ja inclui no androidmanifet  `<application` `android:usesCleartextTraffic="true"/>`
e no manifest tambem ja tem o `<uses-permission android:name="android.permission.INTERNET"/>` 
e nada funciona via wifi, nao importa a rede que eu teste , porem a conexao funciona normal se eu usar a internet do chip.
estou sem saber o que fazer nesse caso.

**Comment by edugemini** [comment]
Olá @30103010, as conexões inseguras (quando você usa http ao invés de https) são desativadas por default no Android e iOS.
A recomendação é não usar http em produção, mas se você só quiser para teste, a solução é:

Se quiser permitir conexões HTTP para compilações de depuração do Android, você pode adicionar o código abaixo em  `$project_path\android\app\src\debug\AndroidManifest.xml`:

```
<application android:usesCleartextTraffic="true"/>
```

Para iOS, você pode seguir [estas instruções](https://docs.flutter.dev/development/add-to-app/ios/project-setup#local-network-privacy-permissions) para criar a propriedade abaixo em Info.plist:
```
<key>NSAppTransportSecurity</key>
<dict>
    <key>NSAllowsArbitraryLoads</key>
    <true/>
</dict>
```

Veja mais detalhes [aqui](https://docs.flutter.dev/release/breaking-changes/network-policy-ios-android)!

**Comment by 30103010** [comment]
oi .. obrigado pela resposta... mais so no wifi que bloqueia, na rede de dados moveis funciona normal... isso que esta me tirando o sono ...kkkkkk
